roles/conf/distsearch/tasks/distributedSearch/server.yml

```diff
@@ -9,11 +9,14 @@
 
 # server
 
+
 - name: "Configure distsearch.conf [distributedSearch] - server state=present"
+  vars:
+    searchpeers: "{{ groups['shcmember'] }}, {{ groups['licensemaster'] }}, {{ groups['masternode'] }}"
   ini_file: dest={{ splunk_conf_path }}/distsearch.conf
             section=distributedSearch
             option=servers
-            value="{{ splunk_distsearch_conf.distributedSearch.servers | create_distsearch_servers  }}"
+            value="{{ searchpeers | create_distsearch_servers  }}"
             state=present
   when: splunk_distsearch_conf.distributedSearch.servers is defined
```

Comment:
to actually address your question...
searchpeers` is comprised of three groups, so instead of having a conditional to check if each of those groups is defined, it just checks that array is defined.

Need to change the var name to `distsearchpeers` so it doesn't get confused with the group `searchpeers`